realm/src/androidTest/java/io/realm/RealmAsyncQueryTests.java

```diff
@@ -2157,6 +2158,67 @@ public void run() {
         executorService.shutdownNow();
     }
 
+    public void testCombiningAsyncAndSync() throws Throwable {
+        final CountDownLatch signalCallbackFinished = new CountDownLatch(1);
+        final Throwable[] threadAssertionError = new Throwable[1];// to catch both Exception & AssertionError
+        final Looper[] backgroundLooper = new Looper[1];
+        ExecutorService executorService = Executors.newSingleThreadExecutor();
+        executorService.submit(new Runnable() {
+            @Override
+            public void run() {
+                Looper.prepare();
+                backgroundLooper[0] = Looper.myLooper();
+
+                Realm realm = null;
+                try {
+                    realm = openRealmInstance("testCombiningAsyncAndSync");
+                    populateTestRealm(realm, 10);
+
+                    Realm.asyncQueryExecutor.pause();
+                    final RealmResults<AllTypes> allTypesAsync = realm.where(AllTypes.class).greaterThan("columnLong", 5).findAllAsync();
+                    final RealmResults<AllTypes> allTypesSync = allTypesAsync.where().greaterThan("columnLong", 3).findAll();
+
+                    assertEquals(0, allTypesAsync.size());
+                    assertEquals(6, allTypesSync.size());
+                    allTypesAsync.addChangeListener(new RealmChangeListener() {
+                        @Override
+                        public void onChange() {
+                            assertEquals(4, allTypesAsync.size());
+                            assertEquals(6, allTypesSync.size());
+                            signalCallbackFinished.countDown();
+                        }
+                    });
+                    Realm.asyncQueryExecutor.resume();
+                    Looper.loop();
+                } catch (Throwable e) {
+                    e.printStackTrace();
+                    threadAssertionError[0] = e;
+
+                } finally {
+                    if (signalCallbackFinished.getCount() > 0) {
+                        signalCallbackFinished.countDown();
+                    }
+                    if (realm != null) {
+                        realm.close();
+                    }
+                }
+            }
+        });
+
+        // wait until the callback of our async query proceed
+        TestHelper.awaitOrFail(signalCallbackFinished, 6000);
+        executorService.shutdownNow();
+        if (null != threadAssertionError[0]) {
+            // throw any assertion errors happened in the background thread
+            throw threadAssertionError[0];
+        }
+        if (backgroundLooper[0] != null) {
+            // failing to quit the looper will not execute the finally block responsible
+            // of closing the Realm
+            backgroundLooper[0].quit();
```

Comment:
This error e.g. showed up when the Nexus S was test device.

1) unit test 1 calls looper.exit()
2) BG looper exist, but hasn't reached `realm.close()` yet on the BG thread.
3) Unit test exists
4) Unit test framework call setup for next unit test
5) Setup method fails because it tries to delete a open Realm (the BG one)
6) Old unit test BG thread closes the Realm (to late)
7) Result is "Cannot delete open Realm" in the log